native/src/main/java/io/ballerina/stdlib/ftp/server/FtpListenerHelper.java

```diff
@@ -74,6 +84,21 @@ public static Object register(BObject ftpListener, BObject service) {
                 FtpConstants.FTP_SERVER_CONNECTOR);
         FtpListener listener = ftpConnector.getFtpListener();
         listener.addService(service);
+
+        Optional<MethodType> methodType = getOnFileChangeMethod(service);
+        if (methodType.isPresent() && methodType.get().getParameters().length == 2) {
+            if (listener.getCaller() == null) {
+                BMap serviceEndpointConfig  = (BMap) ftpListener.getNativeData(FTP_SERVICE_ENDPOINT_CONFIG);;
+                BObject caller = createCaller(serviceEndpointConfig);
+                if (caller instanceof BError) {
+                    BError endpointError = (BError) caller;
+                    return FtpUtil.createError(endpointError.getMessage(), endpointError.getDetails().toString(),
+                            Error.errorType());
+                } else {
+                    listener.setCaller(caller);
+                }
+            }
+        }
         return null;
```

Comment:
We check the negative case first to reduce the white-spaces used for indentation. Check for other places as well.
```suggestion
        if (methodType.isEmpty() || methodType.get().getParameters().length != 2) {
            return null;
        }
        if (listener.getCaller() != null) {
            return null;
        }
        BMap serviceEndpointConfig  = (BMap) ftpListener.getNativeData(FTP_SERVICE_ENDPOINT_CONFIG);;
        BObject caller = createCaller(serviceEndpointConfig);
        if (caller instanceof BError) {
            BError endpointError = (BError) caller;
            return FtpUtil.createError(endpointError.getMessage(), endpointError.getDetails().toString(),
                    Error.errorType());
        } else {
            listener.setCaller(caller);
        }
```

Reply:
Updated